reana_client/utils.py

```diff
@@ -127,3 +130,104 @@ def _validate_reana_yaml(reana_yaml):
         logging.info('Invalid REANA specification: {error}'
                      .format(error=e.message))
         raise e
+
+
+def get_current_click_command(ctx):
+    """Return command (and subcommands) that were used to create given context.
+
+    For example, a context that was created by a command
+    '<executable name> code upload', would return 'code.upload'.
+
+    """
+    command = []
+    while ctx.parent:
+        command.append(ctx.info_name)
+        ctx = ctx.parent
+    return '.'.join(str(x) for x in reversed(command))
+
+
+def upload_files_to_server(user, organization, workflow, filenames, client):
+    """Upload files to REANA-Server.
+
+    Shared e.g. by `code upload` and `inputs upload`.g
+
+    :param user: User ID
+    :param organization: Organization ID
+    :param workflow: ID of that Workflow whose workspace should be
+    used to store the files.
+    :param filenames: Absolute filepath(s) of files to be uploaded.
+    :param client: Instance of Bravado Client.
+    """
+    if workflow:
```

Comment:
What do you think about raising `ValueError` with the message you have inside the `else`?

```python
if not workflow:
    raise ValueError('Workflow name must be provided either with `--workflow` option or with `$REANA_WORKON environment variable` ')
```
And let it be catch by the CLI layer so it will format using click superpowers. It sounds a bit weird to use `click` inside the generic `utils.py`. Moreover, the message shouldn't have `cli` "things" like `--workflow`, imagine someone installs `reana-client` to use the Python API and while using `upload_files_to_server` they get `cli` related messages.

Reply:
Agree.

Reply:
@diegodelemos 
I actually did a "click-free" implementation of `upload_files_to_server` in `utils.py`, but then I thought that we might run into similar cli-level code (de)duplication situation in future and introduced a `common.py` that is under `cli-module`.
Basically this file would include stuff that multiple cli commands use.
Tell me what you think and if it is not a good idea I will change the PR to use "click-free" implementation of `upload_files_to_server`.